libs/pulumi-static-site/src/bucket.ts

```diff
@@ -0,0 +1,160 @@
+import * as aws from '@pulumi/aws'
+import * as pulumi from '@pulumi/pulumi'
+
+export class Bucket extends pulumi.ComponentResource {
+    readonly name: string
+    readonly bucket: aws.s3.Bucket
+
+    constructor(
+        name: string,
+        args: {
+            website?: aws.s3.BucketArgs['website']
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('swm:pulumi-static-site:bucket/Bucket', name, {}, opts)
+
+        this.name = name
+
+        this.bucket = new aws.s3.Bucket(
```

Comment:
This bucket isn't configured so only the CDN can fetch objects? If we are creating a resource for S3 + CloudFront then I think only the cloudfront should be able to get objects from the bucket?

Reply:
Yep, that's a tradeoff to allow the redirects to work, which requires the S3 website endpoints, which require public read.